psi4/src/psi4/libmints/twobody.cc

```diff
@@ -129,13 +192,31 @@ bool TwoBodyAOInt::shell_significant_csam(int M, int N, int R, int S) {
     // Square of Eq. 11
     double mnrs_2 = mn_mn * rs_rs * csam_2;
 
-    return std::abs(mnrs_2) >= screening_threshold_squared_;
+    bool dens = true;
+
+    if (do_dens_screen_) dens = shell_significant_density(M, N, R, S);
+
+    return (std::abs(mnrs_2) >= screening_threshold_squared_ && dens);
 }
 
 bool TwoBodyAOInt::shell_significant_schwarz(int M, int N, int R, int S) {
-    return shell_pair_values_[N * nshell_ + M] * shell_pair_values_[R * nshell_ + S] >= screening_threshold_squared_;
+
+    bool dens = true;
+
+    if (do_dens_screen_) dens = shell_significant_density(M, N, R, S);
+
+    return (shell_pair_values_[N * nshell_ + M] * shell_pair_values_[R * nshell_ + S] >= screening_threshold_squared_ && dens);
```

Comment:
That is, here: this would be just `screening_threshold_` instead of `screening_threshold_squared_`

Reply:
I think you are right, a lot fewer integrals are being computed, but the time has not gone down

Reply:
I mean once you account for the square root in the array